rpc/server.go

```diff
@@ -43,67 +42,71 @@ type Config struct {
 
 // NewServer ...
 func NewServer(cfg *Config) (*Server, error) {
-	s := rpc.NewServer()
-	s.RegisterCodec(NewCodec(), "application/json")
+	rpcServer := rpc.NewServer()
+	rpcServer.RegisterCodec(NewCodec(), "application/json")
 
 	ns := NewNetService(cfg.Net, cfg.XMRTaker, cfg.XMRMaker, cfg.ProtocolBackend.SwapManager())
-	if err := s.RegisterService(ns, "net"); err != nil {
+	if err := rpcServer.RegisterService(ns, "net"); err != nil {
 		return nil, err
 	}
 
-	if err := s.RegisterService(NewPersonalService(cfg.XMRMaker, cfg.ProtocolBackend), "personal"); err != nil {
+	if err := rpcServer.RegisterService(NewPersonalService(cfg.XMRMaker, cfg.ProtocolBackend), "personal"); err != nil {
 		return nil, err
 	}
 
-	if err := s.RegisterService(NewSwapService(cfg.ProtocolBackend.SwapManager(), cfg.XMRTaker, cfg.XMRMaker, cfg.Net), "swap"); err != nil { //nolint:lll
+	if err := rpcServer.RegisterService(NewSwapService(cfg.ProtocolBackend.SwapManager(), cfg.XMRTaker, cfg.XMRMaker, cfg.Net), "swap"); err != nil { //nolint:lll
 		return nil, err
 	}
 
-	return &Server{
-		s:        s,
-		wsServer: newWsServer(cfg.Ctx, cfg.ProtocolBackend.SwapManager(), ns, cfg.ProtocolBackend, cfg.XMRTaker),
-		port:     cfg.Port,
-		wsPort:   cfg.WsPort,
-	}, nil
-}
-
-// Start starts the JSON-RPC server.
-func (s *Server) Start() <-chan error {
-	errCh := make(chan error)
+	wsServer := newWsServer(cfg.Ctx, cfg.ProtocolBackend.SwapManager(), ns, cfg.ProtocolBackend, cfg.XMRTaker)
 
-	go func() {
-		r := mux.NewRouter()
-		r.Handle("/", s.s)
-
-		headersOk := handlers.AllowedHeaders([]string{"content-type", "username", "password"})
-		methodsOk := handlers.AllowedMethods([]string{"GET", "HEAD", "POST", "PUT", "OPTIONS"})
-		originsOk := handlers.AllowedOrigins([]string{"*"})
+	lc := net.ListenConfig{}
+	ln, err := lc.Listen(cfg.Ctx, "tcp", cfg.Address)
+	if err != nil {
+		return nil, err
+	}
 
-		log.Infof("starting RPC server on http://127.0.0.1:%d", s.port)
+	r := mux.NewRouter()
+	r.Handle("/", rpcServer)
+	r.Handle("/ws", wsServer)
+
+	headersOk := handlers.AllowedHeaders([]string{"content-type", "username", "password"})
+	methodsOk := handlers.AllowedMethods([]string{"GET", "HEAD", "POST", "PUT", "OPTIONS"})
+	originsOk := handlers.AllowedOrigins([]string{"*"})
+	server := &http.Server{
+		Addr:              ln.Addr().String(),
+		ReadHeaderTimeout: time.Second,
```

Comment:
Interestingly, the linter flagged this as vulnerable to the Slowloris attack (https://www.cloudflare.com/learning/ddos/ddos-attack-tools/slowloris/) without the `ReadHeaderTimeout` value, but as best I can tell the original `ListenAndServe` call we were using would also be vulnerable, but doesn't get flagged by the linter. I guess because there is not API to set it.